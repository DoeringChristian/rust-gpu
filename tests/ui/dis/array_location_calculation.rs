// build-pass
// compile-flags: -C llvm-args=--disassemble-globals

use spirv_std::{
    self as _,
    glam::{DVec3, IVec4, Mat3, Vec4},
};

#[spirv(matrix)]
pub struct Mat4x3 {
    pub col_0: Vec4,
    pub col_1: Vec4,
    pub col_2: Vec4,
}

#[spirv(fragment)]
pub fn main(
    one: [f32; 7],
    two: [f32; 3],
    three: Mat3,
    four: DVec3,
    five: IVec4,
    six: f32,
    seven: Mat4x3,
    eight: u32,
) {
}
