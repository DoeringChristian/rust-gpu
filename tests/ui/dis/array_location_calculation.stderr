OpCapability Float64
OpCapability Int16
OpCapability Int64
OpCapability Int8
OpCapability ShaderClockKHR
OpCapability Shader
OpCapability VulkanMemoryModel
OpExtension "SPV_KHR_shader_clock"
OpExtension "SPV_KHR_vulkan_memory_model"
OpMemoryModel Logical Vulkan
OpEntryPoint Fragment %1 "main" %2 %3 %4 %5 %6 %7 %8
OpExecutionMode %1 OriginUpperLeft
%9 = OpString "$OPSTRING_FILENAME/array_location_calculation.rs"
OpMemberName %10 0 "x_axis"
OpMemberName %10 1 "y_axis"
OpMemberName %10 2 "z_axis"
OpName %10 "spirv_std::glam::core::storage::Columns3<spirv_std::glam::XYZ<f32>>"
OpMemberName %11 0 "0"
OpName %11 "spirv_std::glam::Mat3"
OpName %12 "array_location_calculation::main"
OpName %2 "one"
OpName %3 "two"
OpName %4 "three"
OpName %5 "four"
OpName %6 "five"
OpName %7 "six"
OpName %8 "seven"
OpMemberName %10 0 "x_axis"
OpMemberName %10 1 "y_axis"
OpMemberName %10 2 "z_axis"
OpMemberName %11 0 "0"
OpMemberName %11 0 "0"
OpMemberName %10 0 "x_axis"
OpMemberName %10 1 "y_axis"
OpMemberName %10 2 "z_axis"
OpDecorate %13 ArrayStride 4
OpDecorate %14 ArrayStride 4
OpMemberDecorate %10 0 Offset 0
OpMemberDecorate %10 1 Offset 16
OpMemberDecorate %10 2 Offset 32
OpMemberDecorate %11 0 Offset 0
OpDecorate %2 Location 0
OpDecorate %3 Location 7
OpDecorate %4 Location 10
OpDecorate %5 Location 13
OpDecorate %6 Location 15
OpDecorate %7 Location 16
OpDecorate %8 Location 17
%15 = OpTypeVoid
%16 = OpTypeFloat 32
%17 = OpTypeInt 32 0
%18 = OpConstant  %17  7
%13 = OpTypeArray %16 %18
%19 = OpConstant  %17  3
%14 = OpTypeArray %16 %19
%20 = OpTypeVector %16 3
%10 = OpTypeStruct %20 %20 %20
%11 = OpTypeStruct %10
%21 = OpTypeFloat 64
%22 = OpTypeVector %21 3
%23 = OpTypeInt 32 1
%24 = OpTypeVector %23 4
%25 = OpTypeFunction %15 %13 %14 %11 %22 %24 %16 %17
%26 = OpTypeFunction %15
%27 = OpTypePointer Input %13
%2 = OpVariable  %27  Input
%28 = OpTypePointer Input %14
%3 = OpVariable  %28  Input
%29 = OpTypePointer Input %11
%4 = OpVariable  %29  Input
%30 = OpTypePointer Input %22
%5 = OpVariable  %30  Input
%31 = OpTypePointer Input %24
%6 = OpVariable  %31  Input
%32 = OpTypePointer Input %16
%7 = OpVariable  %32  Input
%33 = OpTypePointer Input %17
%8 = OpVariable  %33  Input
